libqtile/backend/x11/core.py

```diff
@@ -276,6 +286,15 @@ def on_config_load(self, initial) -> None:
 
             self.qtile.manage(win)
 
+    def _on_client_managed(self, window):
+        self.update_client_list(self.qtile.windows_map)
+        self.update_client_stack()
+        window.change_layer()
+
+    def _on_client_killed(self, window):
+        self.update_client_list(self.qtile.windows_map)
+        self.update_client_stack()
```

Comment:
Just noticed that these two functions are always called together. Should we just merge them into one `self.update_client_lists()`?

Reply:
Yes. We can. However, I don't think the logic in `_NET_CLIENT_LIST_STACKING` is correct. I need to look at that again.